maven-wrapper-distribution/src/resources/mvnw

```diff
@@ -237,8 +243,7 @@ else
             fi
             if [ -e "$javaClass" ]; then
                 log " - Running MavenWrapperDownloader.java ..."
-                ("$JAVA_HOME/bin/java" -cp .mvn/wrapper MavenWrapperDownloader "$wrapperUrl" "$wrapperJarPath")
-                [ $? -eq 0 ] || rm -f "$wrapperJarPath"
+                ("$JAVA_HOME/bin/java" -cp .mvn/wrapper MavenWrapperDownloader "$wrapperUrl" "$wrapperJarPath") || rm -f "$wrapperJarPath"
```

Comment:
not sure why the parenthesis are there. Kept it, but can probably be removed.

Reply:
The parentheses will create a sub shell, so that if the java command fails, the script won't end abruptly (unless the curl/wget above).

Reply:
To be coherent, maybe we should rewrite it to the more simple:
```
"$JAVA_HOME/bin/java" -cp .mvn/wrapper MavenWrapperDownloader "$wrapperUrl" "$wrapperJarPath"
```

Reply:
The script won't exit anyway because neither `set -e` nor `set -o pipefail` is set.

Rewriting is not part of this commit. Let's not fix things here which are not broken. We can open an issue for this, though. I just want to make sure the script behaves the same as before.